contrib/kafka/pkg/dispatcher/dispatcher.go

```diff
@@ -258,12 +266,17 @@ func (d *KafkaDispatcher) setConfig(config *multichannelfanout.Config) {
 	d.config.Store(config)
 }
 
-func NewDispatcher(brokers []string, consumerMode cluster.ConsumerMode, logger *zap.Logger) (*KafkaDispatcher, error) {
-
+func NewDispatcher(provisionerConf *controller.KafkaProvisionerConfig, logger *zap.Logger) (*KafkaDispatcher, error) {
 	conf := sarama.NewConfig()
 	conf.Version = sarama.V1_1_0_0
 	conf.ClientID = controller.Name + "-dispatcher"
-	client, err := sarama.NewClient(brokers, conf)
+
+	if provisionerConf.TlsConfig != nil {
```

Comment:
This seems similar to `createKafkaAdminClient` in controller/channel/reconcile.go. Any chance they can be merged or one call the other?

Reply:
Yes, moved it to `pkg/controller/util.go`